components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodCLIService.kt

```diff
@@ -74,29 +75,35 @@ class GitpodCLIService : RestService() {
                 return "url is missing"
             }
 
-            val resolvedUrl = resolveExternalUrl(url)
-
             return withClient(request, context) { project ->
+                var resolvedUrl = url
+                val uri = URI.create(url)
+                val localHostUriMetadata = LocalHostUri.extractLocalHostUriMetaDataForPortMapping(uri)
+
+                if (localHostUriMetadata.isPresent) {
+                    val gitpodPortForwardingService = service<GitpodPortForwardingService>()
```

Comment:
ℹ️ `service<GitpodPortForwardingService>()` needs to be called here, inside the scope of `withClient`. Otherwise, the service is not found because _GitpodPortForwardingService_ is declared as a `client="controller"` in `extensions.xml`.